core/src/main/java/org/apache/calcite/jdbc/CachingCalciteSchema.java

```diff
@@ -117,15 +100,14 @@ private CachingCalciteSchema(@Nullable CalciteSchema parent, Schema schema,
     return this.cache;
   }
 
-  @Override protected @Nullable CalciteSchema getImplicitSubSchema(String schemaName,
-      boolean caseSensitive) {
-    final long now = System.currentTimeMillis();
-    final SubSchemaCache subSchemaCache = implicitSubSchemaCache.get(now);
-    for (String schemaName2
-        : subSchemaCache.names.range(schemaName, caseSensitive)) {
-      return subSchemaCache.cache.getUnchecked(schemaName2);
-    }
-    return null;
+  @Override protected CalciteSchema createSubSchema(Schema schema, String name) {
+    return new CachingCalciteSchema(this, schema, name);
+  }
+
+  @Override protected <S> Lookup<S> decorateLookup(Lookup<S> lookup) {
```

Comment:
I cannot guess what "decorate" means here

Reply:
I renamed it to `enhanceLookup`and changed the documentation in `CalciteSchema`:

The derived class is able to enhance the lookup e.g. by introducing a cache